jujugui/static/gui/src/app/app.js

```diff
@@ -2060,6 +2062,11 @@ YUI.add('juju-gui', function(Y) {
         }
         if (auth) {
           this.get('users')[service] = auth;
+          // If the profile is visible then we want to rerender it with the
+          // updated username.
+          if (rerenderProfile) {
+            this._renderUserProfile();
```

Comment:
It would be nice if the app state was somehow updated instead of having to manually rerender the profile (this bit is my fault). @hatched do you have any ideas about how to do that?